src/sentry/api/event_search.py

```diff
@@ -2344,12 +2355,14 @@ def get_function_alias(field):
     match = FUNCTION_PATTERN.search(field)
     if match is None:
         return field
-    columns = [c.strip() for c in match.group("columns").split(",") if len(c.strip()) > 0]
-    return get_function_alias_with_columns(match.group("function"), columns)
+    function = match.group("function")
+    columns = parse_arguments(match.group("columns"))
+    return get_function_alias_with_columns(function, columns)
 
 
 def get_function_alias_with_columns(function_name, columns):
-    columns = "_".join(columns).replace(".", "_")
+    columns = re.sub("[^\w]+", "_", "_".join(columns))
+    columns = re.sub("_+", "_", columns).strip("_")
```

Comment:
We want to substitute all special characters with `_` so the resulting alias is valid.